### PR TITLE
review feat(model): annotes getter and setter with a CtRole

### PR DESCRIPTION
**Pull request by tdurieux**
This PR is related to #1368 

TODO tests

### CODE REVIEW

**Comment by monperrus** [comment]
per our discussion
* type is for expression type (no comment, no type access)
* no role in reference interfaces
* remove contract all getters annotated 
* split MEMBER in CONSTRUCTOR/FIELD/EXECUTABLE/NESTED_TYPE

**Comment by monperrus** [comment]
@surli OK for merge for me, please check. 

This will probably be adjusted later, but this is a really good start, thanks @tdurieux

File: src/main/java/spoon/reflect/path/CtRole.java [hunk 1]
```diff
@@ -20,61 +20,59 @@
...
+	//IS_UPPER, // in reference only
 	IS_IMPLICIT,
 	IS_DEFAULT,
 	IS_VARARGS,
-	IS_UPPER,
 	DEFAULT_EXPRESSION,
 	THEN,
 	ELSE,
-	PACKAGE,
+	SUB_PACKAGE,
```
Comment (surli) [inline src/main/java/spoon/reflect/path/CtRole.java]: Why not keeping `PACKAGE` here, it seems clearer?

**Comment by monperrus** [comment]
because it is more explicit, the role is to be a subpackage of another one

**Comment by surli** [comment]
> because it is more explicit, the role is to be a subpackage of another one

Ok I saw the usage after asking.

**Pull request merged**